src/Commands/Handler/Internal/ConnectionRequestHandler.cs

```diff
@@ -138,29 +135,18 @@ public static void PrepareWorldLoad(Player newPlayer)
         {
             newPlayer.Status = ClientStatus.Downloading;
 
-            // Open status window
-            ThreadHelper.dispatcher.Dispatch(() =>
-            {
-                ClientJoinPanel clientJoinPanel = UIView.GetAView().FindUIComponent<ClientJoinPanel>("MPClientJoinPanel");
-                if (clientJoinPanel != null)
-                {
-                    clientJoinPanel.isVisible = true;
-                    clientJoinPanel.StartCheck();
-                }
-                else
-                {
-                    clientJoinPanel = (ClientJoinPanel)UIView.GetAView().AddUIComponent(typeof(ClientJoinPanel));
-                }
-                clientJoinPanel.Focus();
-            });
+            MultiplayerManager.Instance.BlockGame(newPlayer.Username);
+            SimulationManager.instance.SimulationPaused = true;
 
             // Inform other clients about the joining client
             Command.SendToOtherClients(new ClientJoiningCommand
             {
-                JoiningFinished = false
+                JoiningFinished = false,
+                JoiningUsername = newPlayer.Username
             }, newPlayer);
-            MultiplayerManager.Instance.GameBlocked = true;
-            SimulationManager.instance.SimulationPaused = true;
+
+            // Wait to get all remaining pakets processed
```

Comment:
Please provide a more elaborate description on why this is needed

Reply:
Done